src/sparkle/application/__init__.py

```diff
@@ -1,68 +1,132 @@
-            config (Config): Configuration object containing
-              application-specific settings.
-            sources (Dict[str, Source]): Dictionary of data sources
-              used in the application, keyed by source name.
-            writers (List[Writer]): List of Writer objects used for
-              writing the processed data to various destinations.
+            env (ExecutionEnvironment): The environment in which the Spark job is running (LOCAL or AWS).
+            spark_config (Dict[str, Any]): Dictionary containing additional Spark configuration settings.
+            config (Config): Configuration object containing application-specific settings.
+            writers (List[Writer]): List of Writer objects used for writing the processed data.
         """
-        self.spark_session = spark_session
+        self.env = env
+        self.spark_config = spark_config
         self.config = config
         self.writers = writers
+        self.spark_session = self.get_spark_session(env)
+
+    def get_spark_session(self, env: ExecutionEnvironment) -> SparkSession:
+        """Create and return a Spark session based on the environment.
+
+        Args:
+            env (ExecutionEnvironment): The environment type (either LOCAL or AWS).
+
+        Returns:
+            SparkSession: The Spark session configured with the appropriate settings.
+        """
+        if env == ExecutionEnvironment.LOCAL:
+            return self._get_local_session()
+        elif env == ExecutionEnvironment.AWS:
+            return self._get_aws_session()
+        else:
+            raise ValueError(f"Unsupported environment: {env}")
+
+    def _get_local_session(self) -> SparkSession:
+        """Create a Spark session for the local environment."""
+        ivy_settings_path = os.environ.get("IVY_SETTINGS_PATH", None)
+        LOCAL_CONFIG = {
+            "spark.sql.extensions": ",".join(_SPARK_EXTENSIONS),
+            "spark.jars.packages": ",".join(_SPARK_PACKAGES),
+            "spark.sql.session.timeZone": "UTC",
+            "spark.sql.catalog.local": "org.apache.iceberg.spark.SparkCatalog",
+            "spark.sql.catalog.local.type": "hadoop",
+            "spark.sql.catalog.local.warehouse": "./tmp/warehouse",
+        }
+
+        spark_conf = SparkConf()
+        for key, value in LOCAL_CONFIG.items():
+            spark_conf.set(key, str(value))
+        for key, value in self.spark_config.items():
+            spark_conf.set(key, str(value))
+
+        spark_session_builder = SparkSession.builder.master("local[*]").appName(
+            "LocalDataProductApp"
+        )
+        for key, value in spark_conf.getAll():
+            spark_session_builder = spark_session_builder.config(key, value)
+
+        return spark_session_builder.getOrCreate()
+
+    def _get_aws_session(self) -> SparkSession:
+        """Create a Spark session for the AWS environment."""
+        try:
+            from awsglue.context import GlueContext
```

Comment:
Importing Glue is also against the idea of sparkle remaining cloud neutral.

Reply:
@shahinism That is true that spark can have special support for glue environment but I am not sure if that is part of core spark! Anyway my whole argument here is not about providing a cloud agnostic interface but it is about cloud agnostic implementation meaning not using any cloud dependent APIs inside the sparkle. My motivation is that since the Controller is closer to the platform it would be nicer to keep all platform implementations close together.

On the other hand, I can see you may wanna try to keep sparkle fully featured framework without Damavand. In that case, this make total sense. Although, complementary recipes is also part of DataChef's genetic designs. As a result, I don't see the fact that some helper functions like this being implemented in Damavand or another complementary framework would be bad idea. At the end, I think the question would be how bad we want one of our products to be independent of our other products.